smt2parser/src/visitors.rs

```diff
@@ -71,7 +74,7 @@ pub trait SExprVisitor<Constant, Symbol, Keyword> {
 }
 
 #[derive(Debug, PartialEq, Eq, Clone, Hash, Serialize, Deserialize)]
-pub enum Index<Symbol> {
+pub enum Index<Symbol = self::Symbol> {
```

Comment:
I would maybe spell out the full path here and remove the aliases `concrete::*` above.